Topics/starting-a-data-project.md

```diff
@@ -0,0 +1,254 @@
+
+
+## Prerequisites
+All the previous Purgatorio guides.
+
+## Time to complete
+10 minutes.
+
+# Index
+- [Collect Information](#Hunting-for-Information)
+- [Building a Knowledge Tree](#Building-a-Knowledge-Tree)
+- [Choosing a Project Structure](#Choosing-a-Project-Structure)
+- [Reproducibility](#Reproducibility)
+- [Versioning](#Versioning)
+- [Documentation](#Documentation)
+- [Prepare to Fail](#Prepare-to-Fail)
+
+Let's dive right in!
+ 
+ 
+### Hunting for Information
+A vital phase when starting a project is to search for information that can help you. These can be of any kind:
+- Tutorials
+- Documentation
+- Existing projects
+- Research Papers
+- ...
+
+This guide will not teach you how to use [Google](https://www.google.com/), because you will already know if you are reading these lines :-). 
+
+By the way, probably not everyone knows about [**these Google tricks**](https://smallbiztrends.com/2019/03/google-tricks.html)...
+
+But there are a few tips that can come in handy when you start "amassing" knowledge that will then come in handy.
+
+#### Understand what you're looking for
+The first thing to do when dealing with a new problem is to make sure you're looking for the right things. Are you sure the problem is called that for example? Image segmentation is different from image classification! [This Google guide](https://developers.google.com/machine-learning/problem-framing/cases) can help you be sure of the name of your problem.
+
+This may seem trivial, but many useful resources are not found because the correct keywords are not typed into the search engine.
+
+#### Don't re-invent the wheel
+Are you sure that someone hasn't already solved your problem? In that case, if you needed it to solve a real problem you'd already have the dish ready, while if you're doing it to learn you have a base from which to start! Also, observing the code of others is very effective for learning.
+In the latter case, it is still advisable to try to re-implement the solution.
+
+To look if someone has solved the same problem the first place to look is [Github](https://github.com/), the platform where every developer puts Open Source code. 
+Another interesting place can be [Kaggle](https://www.kaggle.com/), the site of the Data Science challenges, where thousands of practitioners and experts challenge each other on real problems, and whose works are available in the form of Notebooks.
+
+Let's suppose for example that I want to solve a problem related to time series: I can type on Kaggle "analysis of time series" and I will probably find dozens of Notebooks that show how to solve a similar problem, and from which you can observe the approach. What a great source of inspiration!
+
+Also check out [TensorFlow Hub](https://www.tensorflow.org/hub), [ModelZoo](https://modelzoo.co/) and [Papers with Code](https://paperswithcode.com/). These three platforms are full of pre-trained models that can come in handy, or even solve your problem already! :) 
```

Comment:
👏

Reply:
I have put together a competition related resource to share with all, not just Kaggle competitions but many other, see https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/competitions.md#competitions-ai-ml-dl-ds

Reply:
it would be an overkill for now. Let's think about a guide "how to make data science challenges" :))))